lib/chef/resource/apt_preference.rb

```diff
@@ -22,7 +22,7 @@ class Chef
   class Resource
     class AptPreference < Chef::Resource
       resource_name :apt_preference
-      provides :apt_preference
+      provides :apt_preference, os: "linux", platform_family: "debian" 
```

Comment:
Should I leave this open ended?

Reply:
resource + provider should probably match